src/rgw/rgw_common.h

```diff
@@ -2423,4 +2446,37 @@ int decode_bl(bufferlist& bl, T& t)
   return 0;
 }
 
+static inline void start_trace(span_structure&& ss, Span&& span, req_state* const s, const char* name, bool should_store)
+{
+  #ifdef WITH_JAEGER
+    if(should_store)
+    {
+      Span span;
+      if(s && !s->stack_span.empty()){
+        span = tracer.child_span(name, s->stack_span.top());
+        ss.set_req_state(s);
+        ss.set_span(span);
```

Comment:
if you are setting a span always when a child_span is created, maybe you can move this step to child_span method itself.